api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1850,6 +1850,138 @@ type AzurePlatformSpec struct {
+}
+
+// AzureResourceManagedIdentities contains the client IDs related to the managed identities needed for HCP control plane
+// and data plane components that authenticate with Azure's API.
+type AzureResourceManagedIdentities struct {
+	// ControlPlaneMIs contains the client IDs of all the managed identities on the HCP control plane needing to
+	// authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	ControlPlaneMIs *ControlPlaneManagedIdentities `json:"controlPlaneMIs"`
+
+	// Future placeholder - DataPlaneMIs * DataPlaneManagedIdentities
+}
+
+// ControlPlaneManagedIdentities contains the client IDs of all the managed identities on the HCP control plane needing
+// to authenticate with Azure's API.
+// Managed identity regex pattern is from Microsoft here - https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftmanagedidentity.
+// The format a managed identity should be `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{managedIdentityName}`.
+type ControlPlaneManagedIdentities struct {
+	// AzureCloudProviderMIClientID is the client ID of a pre-existing managed identity ID associated with the azure
+	// cloud provider, aka ccm.
+	//
+	// +kubebuilder:validation:XValidation:rule="size(self.split('/')) == 11 && self.matches('^/subscriptions/.*/resourceGroups/.*/providers/Microsoft.ManagedIdentity/userAssignedIdentities/.*$')",message="AzureCloudProviderMIClientID must be in the format `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{managedIdentityName}`"
+	// +kubeubilder:validation:XValidation:rule="self.split('/')[2].matches('^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$')",message="the subscriptionId in the AzureCloudProviderMIClientID must be a valid UUID. It should be 5 groups of hyphen separated hexadecimal characters in the form 8-4-4-4-12"
+	// +kubebuilder:validation:XValidation:rule="self.split('/')[4].matches('[a-zA-Z0-9-_\\(\\)\\.]{1,90}')",message="The resourceGroupName should be between 1 and 90 characters, consisting only of alphanumeric characters, hyphens, underscores, periods and parenthesis"
+	// +kubebuilder:validation:XValidation:rule="!self.split('/')[4].endsWith('.')",message="the resourceGroupName in the AzureCloudProviderMIClientID must not end with a period (.) character"
+	// +kubebuilder:validation:XValidation:rule="self.split('/')[8].matches('[a-zA-Z0-9][a-zA-Z0-9-_\\.]{3,128}')",message="AzureCloudProviderMIClientID should be between 3 and 128 characters, consisting only of alphanumeric characters, hyphens and underscores and must start with an alphanumeric character"
+	// +kubebuilder:validation:Required
+	// +required
+	AzureCloudProviderMIClientID string `json:"azureCloudProviderMIClientID"`
```

Comment:
you could introduce a new string type with the validation and just use the type for all the fields here instead of repeating the same validation.

Also this validation is very hard to read, can't we replace it with a single regex expression?